### PR TITLE
:bug: Make image build on releases

### LINKED ISSUE #772: With new release automation, container images workflow is no longer trigged by tag creation
After https://github.com/metal3-io/ip-address-manager/pull/734 was merged, release tags are created by this workflow. This means build-images-action is no longer triggering by tag creation, due Github side security precaution to avoid loops. 

We need to fix it. Some ideas:
- `workflow_run` - change the triggering mechanism, issue is how the tag data is passed
- `workflow_dispatch` - maybe we change both a bit so `uses:` can be integrated as part of release

It does not seem that adding more permissions like `actions: write` etc helps.

### PR DESCRIPTION
**Pull request by mquhuy**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #772 

This should be merged after https://github.com/metal3-io/project-infra/pull/927

### CODE REVIEW

File: .github/workflows/build-images-action.yml [hunk 1]
```diff
@@ -10,6 +10,11 @@ on:
     - 'release-*'
     tags:
     - 'v*'
+  workflow_run:
```
Comment (tuminoid) [inline .github/workflows/build-images-action.yml:13]: This triggers the build images action workflow, yes, but it does not pass context why it was triggered, so it does not know that it needs to build tags.

Comment (mquhuy) [reply .github/workflows/build-images-action.yml:13]: Thanks. I've now added https://github.com/metal3-io/project-infra/pull/927 to account for that.
But now I'm wondering if the "Create Release" wf should be made reusable :-? Wdyt about that?

Comment (tuminoid) [reply .github/workflows/build-images-action.yml:13]: You mean moving it to project-infra? Yes it should, but only after we've ironed out the kinks.

File: .github/workflows/build-images-action.yml [hunk 2]
```diff
@@ -6,20 +6,38 @@ permissions:
...
-      QUAY_USERNAME: ${{ secrets.QUAY_USERNAME }}
-      QUAY_PASSWORD: ${{ secrets.QUAY_PASSWORD }}
-      SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
+    runs-on: ubuntu-latest
+    steps:
+    - name: Set ref
+      id: set_ref
+      run: |
+        # Check if release_tag exists, if so, set ref to "refs/heads/${{ release_tag }}", otherwise use github.ref
+        if [[ -n "${{ needs.create-release.outputs.release_tag }}" ]]; then
```
Comment (tuminoid) [inline .github/workflows/build-images-action.yml:27]: What sets this?

Comment (mquhuy) [reply .github/workflows/build-images-action.yml:27]: It's inside the "Create Release" workflow.

Comment (tuminoid) [reply .github/workflows/build-images-action.yml:27]: Yes, but how does it end up being set here? What populates the `needs.` so it can be used here?

Comment (mquhuy) [reply .github/workflows/build-images-action.yml:27]: It's this line: https://github.com/metal3-io/ip-address-manager/blob/main/.github/workflows/release.yaml#L19

**Review by tuminoid — COMMENTED** [review]
/approve

Technically it looks solid, but these inter-action relationships are so flaky, that I guess we'll find out when we cut next beta or RC with the new automation if it does what it we think it should be doing :) 

Thanks @mquhuy !

**Comment by tuminoid** [comment]
/override metal3-centos-e2e-integration-test-main metal3-ubuntu-e2e-integration-test-main

**Review by lentzi90 — COMMENTED** [review]
/lgtm

**Pull request merged**